### PR TITLE
retry test_migrate_external_table on PermissionDenied error

### LINKED ISSUE #2108: Test failure: `test_migrate_external_table`
<details>
<summary>❌ test_migrate_external_table: databricks.sdk.errors.platform.BadRequest: PERMISSION_DENIED: Failed to get credentials: Storage Credential TEST_STORAGE_CREDENTIAL is not configured correctly. Please contact the owner or your account admin to update the configuration. (20.247s)</summary>

```
databricks.sdk.errors.platform.BadRequest: PERMISSION_DENIED: Failed to get credentials: Storage Credential TEST_STORAGE_CREDENTIAL is not configured correctly. Please contact the owner or your account admin to update the configuration.
07:05 DEBUG [databricks.labs.ucx.mixins.fixtures] added make_dbfs_data_copy fixture: dbfs:/mnt/TEST_MOUNT_NAME/a/b/ziYR
[gw2] linux -- Python 3.10.14 /home/runner/work/ucx/ucx/.venv/bin/python
07:05 DEBUG [databricks.labs.ucx.mixins.fixtures] added make_dbfs_data_copy fixture: dbfs:/mnt/TEST_MOUNT_NAME/a/b/ziYR
07:05 INFO [databricks.labs.ucx.mixins.fixtures] Schema hive_metastore.ucx_skmmf: https://DATABRICKS_HOST/explore/data/hive_metastore/ucx_skmmf
07:05 DEBUG [databricks.labs.ucx.mixins.fixtures] added schema fixture: SchemaInfo(browse_only=None, catalog_name='hive_metastore', catalog_type=None, comment=None, created_at=None, created_by=None, effective_predictive_optimization_flag=None, enable_predictive_optimization=None, full_name='hive_metastore.ucx_skmmf', metastore_id=None, name='ucx_skmmf', owner=None, properties=None, schema_id=None, storage_location=None, storage_root=None, updated_at=None, updated_by=None)
07:05 INFO [databricks.labs.ucx.mixins.fixtures] Table hive_metastore.ucx_skmmf.ucx_tzlrw: https://DATABRICKS_HOST/explore/data/hive_metastore/ucx_skmmf/ucx_tzlrw
07:05 DEBUG [databricks.labs.ucx.mixins.fixtures] added table fixture: TableInfo(access_point=None, browse_only=None, catalog_name='hive_metastore', columns=None, comment=None, created_at=None, created_by=None, data_access_configuration_id=None, data_source_format=<DataSourceFormat.CSV: 'CSV'>, deleted_at=None, delta_runtime_properties_kvpairs=None, effective_predictive_optimization_flag=None, enable_predictive_optimization=None, encryption_details=None, full_name='hive_metastore.ucx_skmmf.ucx_tzlrw', metastore_id=None, name='ucx_tzlrw', owner=None, pipeline_id=None, properties={'RemoveAfter': '2024070908'}, row_filter=None, schema_name='ucx_skmmf', sql_path=None, storage_credential_name=None, storage_location='dbfs:/mnt/TEST_MOUNT_NAME/a/b/ziYR', table_constraints=None, table_id=None, table_type=<TableType.EXTERNAL: 'EXTERNAL'>, updated_at=None, updated_by=None, view_definition=None, view_dependencies=None)
07:05 DEBUG [databricks.labs.ucx.mixins.fixtures] added catalog fixture: CatalogInfo(browse_only=False, catalog_type=<CatalogType.MANAGED_CATALOG: 'MANAGED_CATALOG'>, comment='', connection_name=None, created_at=1720508737154, created_by='0a330eb5-dd51-4d97-b6e4-c474356b1d5d', effective_predictive_optimization_flag=None, enable_predictive_optimization=None, full_name='ucx_cnnmk', isolation_mode=<CatalogIsolationMode.OPEN: 'OPEN'>, metastore_id='8952c1e3-b265-4adf-98c3-6f755e2e1453', name='ucx_cnnmk', options=None, owner='0a330eb5-dd51-4d97-b6e4-c474356b1d5d', properties=None, provider_name=None, provisioning_info=None, securable_kind=<CatalogInfoSecurableKind.CATALOG_STANDARD: 'CATALOG_STANDARD'>, securable_type='CATALOG', share_name=None, storage_location=None, storage_root=None, updated_at=1720508737154, updated_by='0a330eb5-dd51-4d97-b6e4-c474356b1d5d')
07:05 INFO [databricks.labs.ucx.mixins.fixtures] Schema ucx_cnnmk.ucx_skmmf: https://DATABRICKS_HOST/explore/data/ucx_cnnmk/ucx_skmmf
07:05 DEBUG [databricks.labs.ucx.mixins.fixtures] added schema fixture: SchemaInfo(browse_only=None, catalog_name='ucx_cnnmk', catalog_type=None, comment=None, created_at=None, created_by=None, effective_predictive_optimization_flag=None, enable_predictive_optimization=None, full_name='ucx_cnnmk.ucx_skmmf', metastore_id=None, name='ucx_skmmf', owner=None, properties=None, schema_id=None, storage_location=None, storage_root=None, updated_at=None, updated_by=None)
07:05 INFO [tests.integration.hive_metastore.test_migrate] dst_catalog=ucx_cnnmk, external_table=hive_metastore.ucx_skmmf.ucx_tzlrw
07:05 INFO [databricks.labs.ucx.mixins.fixtures] Schema hive_metastore.ucx_sy1ts: https://DATABRICKS_HOST/explore/data/hive_metastore/ucx_sy1ts
07:05 DEBUG [databricks.labs.ucx.mixins.fixtures] added schema fixture: SchemaInfo(browse_only=None, catalog_name='hive_metastore', catalog_type=None, comment=None, created_at=None, created_by=None, effective_predictive_optimization_flag=None, enable_predictive_optimization=None, full_name='hive_metastore.ucx_sy1ts', metastore_id=None, name='ucx_sy1ts', owner=None, properties=None, schema_id=None, storage_location=None, storage_root=None, updated_at=None, updated_by=None)
07:05 DEBUG [databricks.labs.ucx.framework.crawlers] [hive_metastore.ucx_sy1ts.tables] fetching tables inventory
07:05 DEBUG [databricks.labs.ucx.framework.crawlers] [hive_metastore.ucx_sy1ts.tables] crawling new batch for tables
07:05 DEBUG [databricks.labs.ucx.hive_metastore.tables] [hive_metastore.ucx_skmmf] listing tables
07:05 DEBUG [databricks.labs.ucx.hive_metastore.tables] [hive_metastore.ucx_skmmf.ucx_tzlrw] fetching table metadata
07:05 DEBUG [databricks.labs.ucx.framework.crawlers] [hive_metastore.ucx_sy1ts.tables] found 1 new records for tables
07:05 DEBUG [databricks.labs.ucx.hive_metastore.table_migrate] Migrating external table hive_metastore.ucx_skmmf.ucx_tzlrw to using SQL query: SYNC TABLE ucx_cnnmk.ucx_skmmf.ucx_tzlrw FROM hive_metastore.ucx_skmmf.ucx_tzlrw;
07:05 ERROR [databricks.labs.blueprint.parallel] migrate tables(TableToMigrate(src=Table(catalog='hive_metastore', database='ucx_skmmf', name='ucx_tzlrw', object_type='EXTERNAL', table_format='CSV', location='dbfs:/mnt/TEST_MOUNT_NAME/a/b/ziYR', view_text=None, upgraded_to=None, storage_properties={'header': 'true'}, is_partitioned=False), rule=Rule(workspace_name='workspace', catalog_name='ucx_cnnmk', src_schema='ucx_skmmf', dst_schema='ucx_skmmf', src_table='ucx_tzlrw', dst_table='ucx_tzlrw', recon_tolerance_percent=0, compare_rows=False)), [], [], False) task failed: PERMISSION_DENIED: Failed to get credentials: Storage Credential TEST_STORAGE_CREDENTIAL is not configured correctly. Please contact the owner or your account admin to update the configuration.
Traceback (most recent call last):
  File "/home/runner/work/ucx/ucx/.venv/lib/python3.10/site-packages/databricks/labs/blueprint/parallel.py", line 158, in inner
    return func(*args, **kwargs), None
  File "/home/runner/work/ucx/ucx/src/databricks/labs/ucx/hive_metastore/table_migrate.py", line 171, in _migrate_table
    return self._migrate_external_table(src_table.src, src_table.rule, grants)
  File "/home/runner/work/ucx/ucx/src/databricks/labs/ucx/hive_metastore/table_migrate.py", line 247, in _migrate_external_table
    self._backend.execute(self._sql_alter_from(src_table, rule.as_uc_table_key, self._ws.get_workspace_id()))
  File "/home/runner/work/ucx/ucx/.venv/lib/python3.10/site-packages/databricks/labs/lsql/backends.py", line 197, in execute
    self._sql.execute(sql, catalog=catalog, schema=schema)
  File "/home/runner/work/ucx/ucx/.venv/lib/python3.10/site-packages/databricks/labs/lsql/core.py", line 243, in execute
    self._raise_if_needed(status)
  File "/home/runner/work/ucx/ucx/.venv/lib/python3.10/site-packages/databricks/labs/lsql/core.py", line 480, in _raise_if_needed
    raise error_class(error_message)
databricks.sdk.errors.platform.BadRequest: PERMISSION_DENIED: Failed to get credentials: Storage Credential TEST_STORAGE_CREDENTIAL is not configured correctly. Please contact the owner or your account admin to update the configuration.
07:05 CRITICAL [databricks.labs.blueprint.parallel] All 'migrate tables' tasks failed!!!
07:05 DEBUG [databricks.labs.ucx.mixins.fixtures] added make_dbfs_data_copy fixture: dbfs:/mnt/TEST_MOUNT_NAME/a/b/ziYR
07:05 INFO [databricks.labs.ucx.mixins.fixtures] Schema hive_metastore.ucx_skmmf: https://DATABRICKS_HOST/explore/data/hive_metastore/ucx_skmmf
07:05 DEBUG [databricks.labs.ucx.mixins.fixtures] added schema fixture: SchemaInfo(browse_only=None, catalog_name='hive_metastore', catalog_type=None, comment=None, created_at=None, created_by=None, effective_predictive_optimization_flag=None, enable_predictive_optimization=None, full_name='hive_metastore.ucx_skmmf', metastore_id=None, name='ucx_skmmf', owner=None, properties=None, schema_id=None, storage_location=None, storage_root=None, updated_at=None, updated_by=None)
07:05 INFO [databricks.labs.ucx.mixins.fixtures] Table hive_metastore.ucx_skmmf.ucx_tzlrw: https://DATABRICKS_HOST/explore/data/hive_metastore/ucx_skmmf/ucx_tzlrw
07:05 DEBUG [databricks.labs.ucx.mixins.fixtures] added table fixture: TableInfo(access_point=None, browse_only=None, catalog_name='hive_metastore', columns=None, comment=None, created_at=None, created_by=None, data_access_configuration_id=None, data_source_format=<DataSourceFormat.CSV: 'CSV'>, deleted_at=None, delta_runtime_properties_kvpairs=None, effective_predictive_optimization_flag=None, enable_predictive_optimization=None, encryption_details=None, full_name='hive_metastore.ucx_skmmf.ucx_tzlrw', metastore_id=None, name='ucx_tzlrw', owner=None, pipeline_id=None, properties={'RemoveAfter': '2024070908'}, row_filter=None, schema_name='ucx_skmmf', sql_path=None, storage_credential_name=None, storage_location='dbfs:/mnt/TEST_MOUNT_NAME/a/b/ziYR', table_constraints=None, table_id=None, table_type=<TableType.EXTERNAL: 'EXTERNAL'>, updated_at=None, updated_by=None, view_definition=None, view_dependencies=None)
07:05 DEBUG [databricks.labs.ucx.mixins.fixtures] added catalog fixture: CatalogInfo(browse_only=False, catalog_type=<CatalogType.MANAGED_CATALOG: 'MANAGED_CATALOG'>, comment='', connection_name=None, created_at=1720508737154, created_by='0a330eb5-dd51-4d97-b6e4-c474356b1d5d', effective_predictive_optimization_flag=None, enable_predictive_optimization=None, full_name='ucx_cnnmk', isolation_mode=<CatalogIsolationMode.OPEN: 'OPEN'>, metastore_id='8952c1e3-b265-4adf-98c3-6f755e2e1453', name='ucx_cnnmk', options=None, owner='0a330eb5-dd51-4d97-b6e4-c474356b1d5d', properties=None, provider_name=None, provisioning_info=None, securable_kind=<CatalogInfoSecurableKind.CATALOG_STANDARD: 'CATALOG_STANDARD'>, securable_type='CATALOG', share_name=None, storage_location=None, storage_root=None, updated_at=1720508737154, updated_by='0a330eb5-dd51-4d97-b6e4-c474356b1d5d')
07:05 INFO [databricks.labs.ucx.mixins.fixtures] Schema ucx_cnnmk.ucx_skmmf: https://DATABRICKS_HOST/explore/data/ucx_cnnmk/ucx_skmmf
07:05 DEBUG [databricks.labs.ucx.mixins.fixtures] added schema fixture: SchemaInfo(browse_only=None, catalog_name='ucx_cnnmk', catalog_type=None, comment=None, created_at=None, created_by=None, effective_predictive_optimization_flag=None, enable_predictive_optimization=None, full_name='ucx_cnnmk.ucx_skmmf', metastore_id=None, name='ucx_skmmf', owner=None, properties=None, schema_id=None, storage_location=None, storage_root=None, updated_at=None, updated_by=None)
07:05 INFO [tests.integration.hive_metastore.test_migrate] dst_catalog=ucx_cnnmk, external_table=hive_metastore.ucx_skmmf.ucx_tzlrw
07:05 INFO [databricks.labs.ucx.mixins.fixtures] Schema hive_metastore.ucx_sy1ts: https://DATABRICKS_HOST/explore/data/hive_metastore/ucx_sy1ts
07:05 DEBUG [databricks.labs.ucx.mixins.fixtures] added schema fixture: SchemaInfo(browse_only=None, catalog_name='hive_metastore', catalog_type=None, comment=None, created_at=None, created_by=None, effective_predictive_optimization_flag=None, enable_predictive_optimization=None, full_name='hive_metastore.ucx_sy1ts', metastore_id=None, name='ucx_sy1ts', owner=None, properties=None, schema_id=None, storage_location=None, storage_root=None, updated_at=None, updated_by=None)
07:05 DEBUG [databricks.labs.ucx.framework.crawlers] [hive_metastore.ucx_sy1ts.tables] fetching tables inventory
07:05 DEBUG [databricks.labs.ucx.framework.crawlers] [hive_metastore.ucx_sy1ts.tables] crawling new batch for tables
07:05 DEBUG [databricks.labs.ucx.hive_metastore.tables] [hive_metastore.ucx_skmmf] listing tables
07:05 DEBUG [databricks.labs.ucx.hive_metastore.tables] [hive_metastore.ucx_skmmf.ucx_tzlrw] fetching table metadata
07:05 DEBUG [databricks.labs.ucx.framework.crawlers] [hive_metastore.ucx_sy1ts.tables] found 1 new records for tables
07:05 DEBUG [databricks.labs.ucx.hive_metastore.table_migrate] Migrating external table hive_metastore.ucx_skmmf.ucx_tzlrw to using SQL query: SYNC TABLE ucx_cnnmk.ucx_skmmf.ucx_tzlrw FROM hive_metastore.ucx_skmmf.ucx_tzlrw;
07:05 ERROR [databricks.labs.blueprint.parallel] migrate tables(TableToMigrate(src=Table(catalog='hive_metastore', database='ucx_skmmf', name='ucx_tzlrw', object_type='EXTERNAL', table_format='CSV', location='dbfs:/mnt/TEST_MOUNT_NAME/a/b/ziYR', view_text=None, upgraded_to=None, storage_properties={'header': 'true'}, is_partitioned=False), rule=Rule(workspace_name='workspace', catalog_name='ucx_cnnmk', src_schema='ucx_skmmf', dst_schema='ucx_skmmf', src_table='ucx_tzlrw', dst_table='ucx_tzlrw', recon_tolerance_percent=0, compare_rows=False)), [], [], False) task failed: PERMISSION_DENIED: Failed to get credentials: Storage Credential TEST_STORAGE_CREDENTIAL is not configured correctly. Please contact the owner or your account admin to update the configuration.
Traceback (most recent call last):
  File "/home/runner/work/ucx/ucx/.venv/lib/python3.10/site-packages/databricks/labs/blueprint/parallel.py", line 158, in inner
    return func(*args, **kwargs), None
  File "/home/runner/work/ucx/ucx/src/databricks/labs/ucx/hive_metastore/table_migrate.py", line 171, in _migrate_table
    return self._migrate_external_table(src_table.src, src_table.rule, grants)
  File "/home/runner/work/ucx/ucx/src/databricks/labs/ucx/hive_metastore/table_migrate.py", line 247, in _migrate_external_table
    self._backend.execute(self._sql_alter_from(src_table, rule.as_uc_table_key, self._ws.get_workspace_id()))
  File "/home/runner/work/ucx/ucx/.venv/lib/python3.10/site-packages/databricks/labs/lsql/backends.py", line 197, in execute
    self._sql.execute(sql, catalog=catalog, schema=schema)
  File "/home/runner/work/ucx/ucx/.venv/lib/python3.10/site-packages/databricks/labs/lsql/core.py", line 243, in execute
    self._raise_if_needed(status)
  File "/home/runner/work/ucx/ucx/.venv/lib/python3.10/site-packages/databricks/labs/lsql/core.py", line 480, in _raise_if_needed
    raise error_class(error_message)
databricks.sdk.errors.platform.BadRequest: PERMISSION_DENIED: Failed to get credentials: Storage Credential TEST_STORAGE_CREDENTIAL is not configured correctly. Please contact the owner or your account admin to update the configuration.
07:05 CRITICAL [databricks.labs.blueprint.parallel] All 'migrate tables' tasks failed!!!
07:05 DEBUG [databricks.labs.ucx.mixins.fixtures] clearing 1 make_dbfs_data_copy fixtures
07:05 DEBUG [databricks.labs.ucx.mixins.fixtures] removing make_dbfs_data_copy fixture: dbfs:/mnt/TEST_MOUNT_NAME/a/b/ziYR
07:05 DEBUG [databricks.labs.ucx.mixins.fixtures] clearing 0 cluster fixtures
07:05 DEBUG [databricks.labs.ucx.mixins.fixtures] clearing 1 catalog fixtures
07:05 DEBUG [databricks.labs.ucx.mixins.fixtures] removing catalog fixture: CatalogInfo(browse_only=False, catalog_type=<CatalogType.MANAGED_CATALOG: 'MANAGED_CATALOG'>, comment='', connection_name=None, created_at=1720508737154, created_by='0a330eb5-dd51-4d97-b6e4-c474356b1d5d', effective_predictive_optimization_flag=None, enable_predictive_optimization=None, full_name='ucx_cnnmk', isolation_mode=<CatalogIsolationMode.OPEN: 'OPEN'>, metastore_id='8952c1e3-b265-4adf-98c3-6f755e2e1453', name='ucx_cnnmk', options=None, owner='0a330eb5-dd51-4d97-b6e4-c474356b1d5d', properties=None, provider_name=None, provisioning_info=None, securable_kind=<CatalogInfoSecurableKind.CATALOG_STANDARD: 'CATALOG_STANDARD'>, securable_type='CATALOG', share_name=None, storage_location=None, storage_root=None, updated_at=1720508737154, updated_by='0a330eb5-dd51-4d97-b6e4-c474356b1d5d')
07:05 DEBUG [databricks.labs.ucx.mixins.fixtures] clearing 0 workspace group fixtures
07:05 DEBUG [databricks.labs.ucx.mixins.fixtures] clearing 0 table fixtures
07:05 DEBUG [databricks.labs.ucx.mixins.fixtures] clearing 1 table fixtures
07:05 DEBUG [databricks.labs.ucx.mixins.fixtures] removing table fixture: TableInfo(access_point=None, browse_only=None, catalog_name='hive_metastore', columns=None, comment=None, created_at=None, created_by=None, data_access_configuration_id=None, data_source_format=<DataSourceFormat.CSV: 'CSV'>, deleted_at=None, delta_runtime_properties_kvpairs=None, effective_predictive_optimization_flag=None, enable_predictive_optimization=None, encryption_details=None, full_name='hive_metastore.ucx_skmmf.ucx_tzlrw', metastore_id=None, name='ucx_tzlrw', owner=None, pipeline_id=None, properties={'RemoveAfter': '2024070908'}, row_filter=None, schema_name='ucx_skmmf', sql_path=None, storage_credential_name=None, storage_location='dbfs:/mnt/TEST_MOUNT_NAME/a/b/ziYR', table_constraints=None, table_id=None, table_type=<TableType.EXTERNAL: 'EXTERNAL'>, updated_at=None, updated_by=None, view_definition=None, view_dependencies=None)
07:05 DEBUG [databricks.labs.ucx.mixins.fixtures] clearing 3 schema fixtures
07:05 DEBUG [databricks.labs.ucx.mixins.fixtures] removing schema fixture: SchemaInfo(browse_only=None, catalog_name='hive_metastore', catalog_type=None, comment=None, created_at=None, created_by=None, effective_predictive_optimization_flag=None, enable_predictive_optimization=None, full_name='hive_metastore.ucx_skmmf', metastore_id=None, name='ucx_skmmf', owner=None, properties=None, schema_id=None, storage_location=None, storage_root=None, updated_at=None, updated_by=None)
07:05 DEBUG [databricks.labs.ucx.mixins.fixtures] removing schema fixture: SchemaInfo(browse_only=None, catalog_name='ucx_cnnmk', catalog_type=None, comment=None, created_at=None, created_by=None, effective_predictive_optimization_flag=None, enable_predictive_optimization=None, full_name='ucx_cnnmk.ucx_skmmf', metastore_id=None, name='ucx_skmmf', owner=None, properties=None, schema_id=None, storage_location=None, storage_root=None, updated_at=None, updated_by=None)
07:05 DEBUG [databricks.labs.ucx.mixins.fixtures] removing schema fixture: SchemaInfo(browse_only=None, catalog_name='hive_metastore', catalog_type=None, comment=None, created_at=None, created_by=None, effective_predictive_optimization_flag=None, enable_predictive_optimization=None, full_name='hive_metastore.ucx_sy1ts', metastore_id=None, name='ucx_sy1ts', owner=None, properties=None, schema_id=None, storage_location=None, storage_root=None, updated_at=None, updated_by=None)
[gw2] linux -- Python 3.10.14 /home/runner/work/ucx/ucx/.venv/bin/python
```

</details>

<sub>Running from [nightly #116](https://github.com/databrickslabs/ucx/actions/runs/9852469030/job/27201066919)</sub>

### PR DESCRIPTION
**Pull request by ericvergnaud**
## Changes
retry test_migrate_external_table on PermissionDenied error

### Linked issues
Resolves #2108

### Functionality
None

### Tests
- [x] manually tested

### CODE REVIEW

**Review by JCZuurmond — COMMENTED** [review]
What causes the `PermissionDenied` to occure?

File: tests/integration/hive_metastore/test_migrate.py [hunk 1]
```diff
@@ -126,7 +126,7 @@ def test_migrate_tables_with_cache_should_not_create_table(
     assert target_tables[0]["tableName"] == table_name
 
 
-@retried(on=[NotFound], timeout=timedelta(minutes=2))
+@retried(on=[NotFound, PermissionDenied], timeout=timedelta(minutes=2))
```
Comment (nfx) [inline tests/integration/hive_metastore/test_migrate.py:129]: what message causes permission denied?

**Review by nfx — CHANGES_REQUESTED** [review]

File: tests/integration/hive_metastore/test_migrate.py (hunk 1 above)
Comment (ericvergnaud) [reply tests/integration/hive_metastore/test_migrate.py:129]: I see this in the logs: 
databricks.sdk.errors.platform.BadRequest: PERMISSION_DENIED: Failed to get credentials: Storage Credential TEST_STORAGE_CREDENTIAL is not configured correctly. Please contact the owner or your account admin to update the configuration.

**Comment by ericvergnaud** [comment]
> What causes the `PermissionDenied` to occure?

I see this in the logs:
databricks.sdk.errors.platform.BadRequest: PERMISSION_DENIED: Failed to get credentials: Storage Credential TEST_STORAGE_CREDENTIAL is not configured correctly. Please contact the owner or your account admin to update the configuration.

**Pull request closed without merging**